src/core/worker.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+#include <atomic>
+#include <queue>
+#include "crypto.hpp"
+#include "api.hpp"
+#include "logger.hpp"
+
+#if WIN32
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+#else
+#include <pthread.h>
+#define THREAD_PRIORITY_TIME_CRITICAL -15
+#define THREAD_PRIORITY_HIGHEST -10
+#define THREAD_PRIORITY_ABOVE_NORMAL -5
+#define THREAD_PRIORITY_NORMAL 0
+#define THREAD_PRIORITY_BELOW_NORMAL 5
+#define THREAD_PRIORITY_IDLE 19
+#endif
+
+struct Job {
+    string host;
+    Block newBlock;
+};
+
+class Worker
+{
+public:
+    static std::atomic<uint32_t> accepted_blocks;
+    static std::atomic<uint32_t> rejected_blocks;
+    static std::atomic<uint32_t> earned;
+
+    Worker(int thread_priority) : thread(Worker::start, this), queue(), qmutex(), cond()
+    {
+        int priority;
+
+        switch (thread_priority) {
+        case 5:
+            priority = THREAD_PRIORITY_TIME_CRITICAL;
+            break;
+        case 4:
+            priority = THREAD_PRIORITY_HIGHEST;
+            break;
+        case 3:
+            priority = THREAD_PRIORITY_ABOVE_NORMAL;
+            break;
+        case 2:
+            priority = THREAD_PRIORITY_NORMAL;
+            break;
+        case 1:
+            priority = THREAD_PRIORITY_BELOW_NORMAL;
+            break;
+        default:
+            priority = THREAD_PRIORITY_IDLE;
+        }
+#if WIN32
+        SetThreadPriority(static_cast<HANDLE>(thread.native_handle()), priority);
+#else
+        int sched;
+
+        if (priority == THREAD_PRIORITY_IDLE) 
+        {
+            sched = SCHED_IDLE;
```

Comment:
I get a build error here:
```
/src/panda-coin/src/core/worker.hpp:67:21: error: use of undeclared identifier 'SCHED_IDLE'
            sched = SCHED_IDLE;
                    ^
```

Reply:
Could you test if it works by adding

`#ifndef SCHED_IDLE
#define SCHED_IDLE	SCHED_OTHER
#endif`

Miner should utilize all cores and you should still have responsive system.

Reply:
I'm compiling on OSX Big Sur. Will try this out